dev/refresh_source.py

```diff
@@ -226,6 +230,18 @@ def git_clone(self, repository, owner=None):
       shell_result = run_and_monitor(
           'git clone {url} -b {branch}'.format(url=origin_url, branch=branch),
           echo=False)
+      if (shell_result.returncode
+          and branch != "master"
+          and self.__options.allow_master):
+        if shell_result.stdout:
```

Comment:
Is there a specific pattern we'd want to look for in the output? This would default to master no matter the failure (e.g. network unavailability, whatever) which may cause some confusion. If we know exactly which message we need to retry on we can check for it here.